modules/imgcodecs/perf/perf_main.cpp

```diff
@@ -8,3 +8,35 @@
 #endif
 
 CV_PERF_TEST_MAIN(imgcodecs)
+
+namespace opencv_test
+{
+
+using namespace perf;
+
+#define TYPICAL_MAT_SIZES_TIFF_DECODE  cv::Size(128, 128), cv::Size(256, 256), cv::Size(512, 512), cv::Size(1024, 1024)
+#define TYPICAL_MAT_TYPES_TIFF_DECODE  CV_8UC1, CV_8UC3, CV_16UC1, CV_16UC3
+#define TYPICAL_MATS_TIFF_DECODE       testing::Combine( testing::Values( TYPICAL_MAT_SIZES_TIFF_DECODE), testing::Values( TYPICAL_MAT_TYPES_TIFF_DECODE) )
+
+PERF_TEST_P(Size_MatType, tiffDecode, TYPICAL_MATS_TIFF_DECODE)
+{
+    cv::Size sz = get<0>(GetParam());
+    int type = get<1>(GetParam());
+
+    cv::Mat src = Mat(sz, type);
+    cv::Mat dst = Mat(sz, type);
+    cv::randu(src, 0, (CV_MAT_DEPTH(type) == CV_16U) ? 65535 : 255);
+    std::vector<uchar> buf;
+    cv::imencode(".tiff", src, buf);
+
+    declare.in(src, WARMUP_RNG).out(dst);
+
+    TEST_CYCLE()
+    {
+      cv::imdecode(buf, cv::IMREAD_UNCHANGED, &dst);
+    }
+
+    SANITY_CHECK(dst);
```

Comment:
OpenCV performance tests usually do not check all results, but control control sums or subset of result data. To have `SANITY_CHECK` that works correctly you need to record reference results to file and then propose it to opencv_extra repository. So, please run the test with `--perf_write_sanity` option. New file `imcodecs.xml` should appear in `opencv_extra/testdata/perf`. Please create PR with the file to extra repository. The pr branch should be named in the same way as your branch in main repo. In this case they are tested together.

Reply:
I will try that

Reply:
Done. But I can't see the perf regression with that system (which does not mean that there is none).
Still investigating to see if I am proceeding correclty.